PurchasesTests/TestHelpers/ObjCThrowExceptionMatcher.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  ObjCThrowExceptionMatcher.swift
+//  PurchasesTests
+//
+//  Created by Joshua Liebowitz on 6/10/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import Nimble
+
+enum ObjCException: String {
+    
+    case parameterAssert = "NSInternalInconsistencyException"
+    
+}
+
+func expectToThrowException(_ named: ObjCException? = nil, closure: @escaping  () -> Void) -> Void {
+    do {
+        try RCObjC.catchException {
+            let _ = closure()
+            fail("No exception thrown.")
+        }
+    } catch {
+        let error = error as NSError
+        print("Threw " + error.domain)
+        let message = messageForError(error: error, named: named?.rawValue)
+        let matches = errorMatchesNonNilFields(error, named: named?.rawValue)
+        if !matches {
+            Nimble.fail(message.expectedMessage)
+        }
+    }
+}
+
+func expectToNotThrowException(closure: @escaping  () -> Void) -> Void {
+    do {
+        try RCObjC.catchException {
+            let _ = closure()
+        }
+    } catch {
+        let error = error as NSError
+        print("Threw " + error.domain)
+        let message = messageForError(error: error, named: error.domain)
+        Nimble.fail(message.expectedMessage)
+    }
+}
+
+func messageForError(error: NSError?, named: String?) -> ExpectationMessage {
```

Comment:
let's make these `private`, since they're only meant to be inside this file but they're global

Reply:
✅